controllers/snapshot/snapshot_adapter.go

```diff
@@ -446,3 +545,98 @@ func (a *Adapter) updateExistingSnapshotEnvironmentBindingWithSnapshot(snapshotE
 
 	return snapshotEnvironmentBinding, nil
 }
+
+// createCopyOfExistingEnvironment uses existing env as input, specifies namespace where the environment is situated,
+// integrationTestScenario contains information about existing environment
+// snapshot is mainly used for adding labels
+// returns copy of already existing environment with updated envVars
+func (a *Adapter) createCopyOfExistingEnvironment(existingEnvironment *applicationapiv1alpha1.Environment, namespace string, integrationTestScenario *v1alpha1.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot, application *applicationapiv1alpha1.Application) (*applicationapiv1alpha1.Environment, error) {
+	// Try to find a available DeploymentTargetClass with the right provisioner
+	deploymentTargetClass, err := a.FindAvailableDeploymentTargetClass()
+	if err != nil || deploymentTargetClass == nil {
+		a.logger.Error(err, "Failed to find deploymentTargetClass with right provisioner for copy of existingEnvironment!",
+			"existingEnvironment.NameSpace: ", existingEnvironment.Namespace,
+			"existingEnvironment.Name: ", existingEnvironment.Name,
+			"deploymentTargetClass.Provisioner: ", applicationapiv1alpha1.Provisioner_Devsandbox)
+		return nil, err
```

Comment:
This is not nice, `err` may be `nil` because `FindAvailableDeploymentTargetClass` may return `nil, nil` when targetClass doesn't exist. We should construct new error message when `deploymentTargetClass == nil`

Reply:
Yes, this is why I add `if err != nil || deploymentTargetClass == nil` when calling it in https://github.com/redhat-appstudio/integration-service/pull/144/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R556.

Let me also add some more info to `err`

Reply:
Add more info to `Err`

Reply:
yes, but if `err == nil` and `deploymentTargetClass == nil`, you will get get following error message

```
Failed to find deploymentTargetClass with right provisioner for copy of existingEnvironment env: nil
```

and I'm not sure if nil will work with `%w` formatting string used for error chaining